pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -165,28 +165,31 @@ private TransferableBlock produceAggregatedBlock() {
    */
   private boolean consumeInputBlocks() {
     TransferableBlock block = _inputOperator.nextBlock();
-    // setting upstream error block
-    if (block.isErrorBlock()) {
-      _upstreamErrorBlock = block;
-      return true;
-    } else if (block.isEndOfStreamBlock()) {
-      _readyToConstruct = true;
-      return true;
-    }
+    while (!block.isNoOpBlock()) {
```

Comment:
this looks like a tight loop?

Reply:
what do you mean by this? the last line of the loop gets the next block from upstream

Reply:
right. i thought the comment suggest we should put the opChain back into the queue until the next available mail triggers it again?

Reply:
oh i see so the idea is to process as much block as possible until the queue is empty then put back into the opChain ?

Reply:
yup! that makes the scheduling logic much simpler because as soon as I schedule I know that I have no more mail I need to hold on to